app/pages/settings/email.spec.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import assert from 'assert';
+import { mount } from 'enzyme';
+import talkClient from 'panoptes-client/lib/talk-client';
+import EmailSettings from './email';
+
+const subscriptionPreferences = [
+  {
+    id: 1,
+    category: 'participating_discussions',
+    email_digest: 'daily'
+  },
+  {
+    id: 2,
+    category: 'followed_discussions',
+    email_digest: 'daily'
+  },
+  {
+    id: 3,
+    category: 'mentions',
+    email_digest: 'immediate'
+  },
+  {
+    id: 4,
+    category: 'group_mentions',
+    email_digest: 'immediate'
+  },
+  {
+    id: 5,
+    category: 'messages',
+    email_digest: 'never'
+  },
+  {
+    id: 6,
+    category: 'started_discussions',
+    email_digest: 'weekly'
+  }
+];
+
+talkClient.type = () => {
+  return {
+    get() {
+      return Promise.resolve(subscriptionPreferences);
+    }
+  };
+};
+
+const project = {
+  display_name: 'A test project',
+  title: 'A test project'
+};
+
+const anotherProject = {
+  display_name: 'Another test project',
+  title: 'Another test project'
+};
+
+const preferences = [
+  {
+    email_communication: true,
+    get() {
+      return Promise.resolve(project);
+    },
+    getMeta() {
+      return {};
+    }
+  },
+  {
+    email_communication: false,
+    get() {
+      return Promise.resolve(anotherProject);
+    },
+    getMeta() {
+      return {};
+    }
+  }
+];
+
+const user = {
+  email: 'An email address',
+  beta_email_communication: false,
+  global_email_communication: true,
+  project_email_communication: true,
+  get() {
+    return Promise.resolve(preferences);
+  }
+};
+
+describe('EmailSettings', () => {
+  const wrapper = mount(<EmailSettings user={user} />);
+
+  beforeEach(() => wrapper.update());
+
+  it('renders the email address', () => {
+    const email = wrapper.find('input[name="email"]');
+    assert.equal(email.prop('value'), user.email);
+  });
+
+  it('shows global email preference correctly', () => {
+    const projectEmail = wrapper.find('input[name="global_email_communication"]');
+    assert.equal(projectEmail.prop('checked'), user.project_email_communication);
+  });
```

Comment:
I think `user.project_email_communication` should be `user.global_email_communication` here.

Reply:
Good catch!